spec/editors_draft.html

```diff
@@ -3298,6 +3298,231 @@ <h2>Evaluation Semantics</h2>
+      </p>
+        
+      <p>
+        The following sections describe requirements for conforming SPARQL systems that must be applied to composite values during their import into and export from datasets to support proper handling of blank nodes.
+      </p>
+    </section>
+
+    <section>
+      <h2>Importing Requirements</h2>
+      <p>
+        Conforming implementations MUST process <a>cdt:List literal</a>s and <a>cdt:Map literal</a>s during loading, replacing the substrings containing the serialized form of any blank nodes with identifier <var>id</var> with a blank node label <var>b</var>, such that
+      </p>
+      
+      <ul>
+        <li>within the scope of the document being processed, all occurrences of <var>id</var> are replaced with the same <var>b</var> value, and</li>
+        <li>within the scope of the document being processed, two distinct <var>id</var> values are replaced with distinct <var>b</var> values, and</li>
+        <li>
+        	if the document being processed has a mechanism for representing blank nodes outside of composite values (e.g. `_:b1` syntax in N-Triples or Turtle, `rdf:nodeID` in RDF/XML),
+        	and the blank node identifier <var>id</var> appears both inside and outside of composite values,
+        	and the parser emits a blank node <var>B</var> for occurrences of <var>id</var> outside of composite values,
+        	then the rewritten blank node label <var>b</var> MUST be a lexical form that maps to <var>B</var> in the lexical-to-value mapping (that is, <i><a>bnl2bn</a></i>(<var>id</var>) = <var>B</var>).</li>
+      </ul>
+
+      <p>
+        Here, "loading" is any single process by which composite values enter a SPARQL system or RDF Dataset. This includes (but is not limited to):
+      </p>
+
+      <ul>
+        <li>Parsing literal values in a [[SPARQL12-QUERY]] or [[SPARQL12-UPDATE]] string</li>
+        <li>The loading of RDF data via a single [[SPARQL12-GRAPH-STORE-PROTOCOL]] request</li>
+        <li>The loading of RDF data via [[SPARQL12-UPDATE]] `LOAD` operation</li>
+        <li>The loading of SPARQL Results via [[SPARQL11-FEDERATED-QUERY]]</li>
+        <li>The loading of RDF data via any single [[SPARQL11-QUERY]] `FROM` or `FROM NAMED` clause in systems <a data-cite="SPARQL11-SERVICE-DESCRIPTION#sd-dereferencesuris">supporting the dereferencing and loading of RDF data in the construction of the dataset</a></li>
+        <li>Importing of any RDF data to the underlying RDF Dataset (via means other than [[SPARQL11-UPDATE]] or [[SPARQL12-GRAPH-STORE-PROTOCOL]])</li>
+      </ul>
+
+      <p>Loading the Turtle in <a href="#load-renaming-blank-example-simple"></a>, data with `_:b0` in a <a>cdt:List literal</a> MUST result in an (internal) dataset with a single triple whose object is a <a>cdt:List literal</a> containing a blank node whose identifier is NOT "b0" (<a href="#load-renaming-blank-example-simple-result"></a>):
```

Comment:
Agreed on the first point. Will update the language.

Regarding the second point, that's fair. Given the current requirements in the bullet points at the beginning of the section, I guess it doesn't *have* to be the case that it's a different value after the rewriting. But such a situation would be problematic if you loaded the same data twice, you'd need it to be different from `b0` in at least one of those rewritings. We could simply require that the rewriting produce some new value that is distinct from the original value. I think that would resolve the problem, though it feels a bit inelegant. Any thoughts on other solutions?

Reply:
I don't think the problem can be resolved by simply requiring that the rewriting produces some new value (blank node identifier) distinct from the original value (i.e., the blank node identifier given in the lexical form of the cdt:List literal. The crux is that the new blank node identifier would also need to be different from any blank node identifier that was used before when loading this file. Perhaps a way to tie this requirement into this section (in particular also into the three bullet points at the beginning of this section), is to say that the new blank node identifier _b_ must be selected such that the blank node _B_ = _bnl2bn_(_b_) is not currently used within the system.

Reply:
> Perhaps a way to tie this requirement into this section (in particular also into the three bullet points at the beginning of this section), is to say that the new blank node identifier b must be selected such that the blank node B = bnl2bn(b) is not currently used within the system.

Yes, exactly. I just wrote new language for the import requirements while addressing a different PR comment:

> bnl2bn(b) must be a fresh blank node not already used in the SPARQL system or RDF Dataset

I suspect the "NOT 'b0'" language is still too strong, and wonder if it can simply be removed now. Thoughts?

Reply:
> I suspect the "NOT 'b0'" language is still too strong, and wonder if it can simply be removed now. Thoughts?

Yes, I think that the part about "NOT 'b0'" should be removed. Yet, after removing it, it becomes a bit unclear what point the example is trying to make. Perhaps the whole example should be removed?